### PR TITLE
Add a spec for Restart test

### LINKED ISSUE #9964: Create a `restart` HapiSpecOp that will freeze and start all `HapiTestEngine` nodes

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #9964 

- Added a test suite that does submit mixed operations, freeze, stops nodes, restarts nodes, waits for ACTIVE status, and then submits some more mixed operations.
- Updated all Service schemas to use Current Version instead of Genesis version

**Testing:**
1. In `swirlds.log` I could see 
   - Platform freezes with following log 
```
2023-11-26 22:52:03.830 545      INFO  PLATFORM_STATUS  <<platform: status-state-machine>> PlatformStatusStateMachine: Platform spent 34.7 s in ACTIVE. Now in FREEZING {"oldStatus":"ACTIVE","newStatus":"FREEZING"} [com.swirlds.logging.legacy.payload.PlatformStatusPayload]
2023-11-26 22:52:03.841 547      INFO  FREEZE           <<platform-core: thread-cons 0>> DualStateImpl: setLastFrozenTimeToBeCurrentFreezeTime: 2023-11-27T04:52:03.597263Z
2023-11-26 22:52:03.855 548      INFO  FREEZE           <<platform-core: thread-cons 0>> DualStateImpl: Set last frozen time {"lastFrozenTime":"2023-11-27T04:52:03.597263Z"} [com.swirlds.logging.legacy.payload.SetLastFrozenTimePayload]
2023-11-26 22:52:03.898 550      INFO  STATE_TO_DISK    <<platform-core: thread-curr 0>> SavedStateController: Signed state from round 255 created, will eventually be written to disk, for reason: FREEZE_STATE
2023-11-26 22:52:03.902 551      INFO  STATE_TO_DISK    <<scheduler signed_state_file_manager>> SignedStateFileWriter: Started writing round 255 state to disk. Reason: FREEZE_STATE, directory: /Users/neeharikasompalli/Documents/Hedera/Repos/hedera-services/hedera-node/test-clients/build/hapi-test/node0/data/saved/com.hedera.services.ServicesMain/0/123/255
2023-11-26 22:52:06.302 560      INFO  STATE_TO_DISK    <<scheduler signed_state_file_manager>> SignedStateFileWriter: Information for state written to disk:
```
-  Then writes saved state and shuts down
```
2023-11-26 22:52:06.677 566      INFO  PLATFORM_STATUS  <<platform: status-state-machine>> PlatformStatusStateMachine: Platform spent 2.8 s in FREEZING. Now in FREEZE_COMPLETE {"oldStatus":"FREEZING","newStatus":"FREEZE_COMPLETE"} [com.swirlds.logging.legacy.payload.PlatformStatusPayload]
2023-11-26 22:52:08.931 567      INFO  STARTUP          <<browser: shutdown-hook>> Log4jSetup: JVM is shutting down.
```
- Loads last written state
```
2023-11-26 22:52:39.783 25       INFO  STARTUP          <main> StartupStateUtils: Platform has loaded a saved state {"round":255,"consensusTimestamp":"2023-11-27T04:52:03.598460Z"} [com.swirlds.logging.legacy.payload.SavedStateLoadedPayload]
2023-11-26 22:52:39.807 26       INFO  STARTUP          <main> BootstrapUtils: Not upgrading software, current software is version HederaSoftwareVersion{hapiVersion=v0.44.0, servicesVersion=v0.45.0-SNAPSHOT}.
2023-11-26 22:52:39.808 27       INFO  STARTUP          <main> AddressBookInitializer: Using the loaded state's address book and weight values.
```
- Turns ACTIVE
```
2023-11-26 22:52:53.040 50       INFO  PLATFORM_STATUS  <<platform: status-state-machine>> PlatformStatusStateMachine: Platform spent 868.0 ms in CHECKING. Now in ACTIVE {"oldStatus":"CHECKING","newStatus":"ACTIVE"} [com.swirlds.logging.legacy.payload.PlatformStatusPayload]
```

**Client Log**:
Submits all transactions after restart successfully 

**NOTE:**
- When the platform loads state, the ` BlockRecordService.BLOCKS` hash is different from what is written while freezing. Still need to Debug why that happens.

### CODE REVIEW

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/Service.java [hunk 1]
```diff
@@ -28,6 +29,7 @@
  * crypto-service, token-service etc.,
  */
 public interface Service {
+    SemanticVersion CURRENT_VERSION = new SemanticVersion(0, 45, 0, "SNAPSHOT", "");
```
Comment (mhess-swl) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/Service.java:32]: We're not permanently using this version, are we? The testnet and mainnet migrations will need to do a genesis initialization

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/Service.java:32]: Why will testnet and mainnet do genesis migration ?

Comment (mhess-swl) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/Service.java]: There _may_ be a Schema for each version, but it's not required. The thing about the schemas is _each schema assumes that the migrations for all previous schemas have already been done_

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/Service.java]: yes, but my question is should we keep all the schemas for every migration we add in repo.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/BlockRecordService.java [hunk 2]
```diff
@@ -68,14 +68,20 @@ public Set<StateDefinition> statesToCreate() {
...
+                    blocksState.put(blocks);
+                    final var runningHashes =
+                            RunningHashes.newBuilder().runningHash(GENESIS_HASH).build();
+                    runningHashState.put(runningHashes);
+                }
+            }
+
+            @Override
+            public void restart(@NonNull final MigrationContext ctx) {
+                // TODO : seems there should be something done here to match hash of the state after loading on restart
```
Comment (tinker-michaelj) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/BlockRecordService.java:84]: Very interesting. 🤔

**Review by tinker-michaelj — APPROVED** [review]
This all makes sense to me, ty @Neeharika-Sompalli ! 🥂

**Pull request merged**